File: leetcode/python/2482_difference_between_ones_and_zeros_in_row_and_column.py
class Solution:
    def onesMinusZeros(self, grid: list[list[int]]) -> list[list[int]]:
        n: int = len(grid)
        m: int = len(grid[0])
        row: list[int] = [0] * n
        col: list[int] = [0] * m
        for i in range(n):
            for j in range(m):
                if grid[i][j]:
                    row[i] += 1
                    col[j] += 1
                    
                    
        res = [[0] * m for _ in range(n)]
        for i in range(n):
            for j in range(m):
                res[i][j] += row[i] + col[j]
                res[i][j] -= (n - row[i]) + (m - col[j])
        return res
                
        

File: leetcode/rust/src/2482_difference_between_ones_and_zeros_in_row_and_column.rs
impl Solution {
    pub fn ones_minus_zeros(grid: Vec<Vec<i32>>) -> Vec<Vec<i32>> {
        let m = grid.len();
        let n = grid[0].len();
        let mut one_rows = vec![0; m];
        let mut one_cols = vec![0; n];

        for i in 0..m {
            for j in 0..n {
                if grid[i][j] == 1 {
                    one_rows[i] += 1;
                    one_cols[j] += 1;
                }
            }
        }

        let mut res: Vec<Vec<i32>> = vec![vec![0; n]; m];
        for i in 0..m {
            for j in 0..n {
                res[i][j] += one_rows[i] + one_cols[j];
                res[i][j] -= (m as i32 - one_rows[i]) + (n as i32 - one_cols[j]);
            }
        }
        res

    }
}